plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerLabelProvider.java

```diff
@@ -261,4 +262,16 @@ public void applyStyles(TextStyle textStyle) {
 		});
 		return value;
 	}
+
+	private static final OpenShiftExplorerLabelProvider instanceForDescription = new OpenShiftExplorerLabelProvider();
+	static {
+		//Status line has with CLabel its own ellipses adjusting string to available space.
+		//We could set Integer.MAX_VALUE as limit, but performance will be too low for really long texts.
+		instanceForDescription.setLabelLimit(512);
```

Comment:
I'd instantiate the label provider in the static block, too since we dont get around this static initializer.
I'd also extract the 512 to a constant for enhanced code readability